snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/events/TrackerError.java

```diff
@@ -0,0 +1,99 @@
+package com.snowplowanalytics.snowplow.tracker.events;
+
+import com.snowplowanalytics.snowplow.tracker.constants.Parameters;
+import com.snowplowanalytics.snowplow.tracker.constants.TrackerConstants;
+import com.snowplowanalytics.snowplow.tracker.payload.Payload;
+import com.snowplowanalytics.snowplow.tracker.payload.SelfDescribingJson;
+import com.snowplowanalytics.snowplow.tracker.payload.TrackerPayload;
+import com.snowplowanalytics.snowplow.tracker.utils.Preconditions;
+import com.snowplowanalytics.snowplow.tracker.utils.Util;
+
+public class TrackerError extends AbstractEvent {
+    private static final int MAX_MESSAGE_LENGTH = 2048;
+    private static final int MAX_STACK_LENGTH = 8096;
+    private static final int MAX_EXCEPTION_NAME_LENGTH = 1024;
+
+    private final String source;
+    private final String message;
+    private final Throwable throwable;
+
+    public static abstract class Builder<T extends Builder<T>> extends AbstractEvent.Builder<T> {
+        private String source;
+        private String message;
+        private Throwable throwable;
+
+        public T source(String source) {
+            this.source = source;
+            return self();
+        }
+
+        public T message(String message) {
+            this.message = message;
+            return self();
+        }
+
+        public T throwable(Throwable throwable) {
+            this.throwable = throwable;
+            return self();
+        }
+
+        public TrackerError build() {
+            return new TrackerError(this);
+        }
+    }
+
+    private static class Builder2 extends Builder<Builder2> {
+        @Override
+        protected Builder2 self() {
+            return this;
+        }
+    }
+
+    public static Builder<?> builder() {
+        return new Builder2();
+    }
+
+    private TrackerError(Builder<?> builder) {
+        super(builder);
+
+        // Precondition checks
+        Preconditions.checkNotNull(builder.source);
+        Preconditions.checkNotNull(builder.message);
+
+        this.source = builder.source;
+        this.message = builder.message;
+        this.throwable = builder.throwable;
+    }
+
+    @Override
+    public Payload getPayload() {
+        return new SelfDescribingJson(TrackerConstants.APPLICATION_ERROR_SCHEMA, getData());
```

Comment:
I'm in two minds as to whether this is the correct schema for this or if we should create a new one.
The application_crash schema was created to report when applications that contain the tracker throw unhandled exceptions, however in this case we are allowing customers to turn on diagnostics that allow the tracker to emit log events - the application hasn't crashed at all and this could inflate users crash statistics if they are counting this event type.
I wonder if a new iglu central schema would be better and more descriptive here, perhaps `tracker_log` or `tracker_diagnostic`?

Reply:
I would go with `tracker_diagnostic` 👍